kubeprod/pkg/gke/platform.go

```diff
@@ -283,7 +284,9 @@ func (conf *GKEConfig) Generate(ctx context.Context) error {
 				Do()
 			if err != nil {
 				// FIXME: retry on etag conflict
-				return fmt.Errorf("failed to set IAM policy on project %q: %v", conf.ExternalDNS.Project, err)
+				return fmt.Errorf("failed to set IAM policy on project %q: user lacks "+
```

Comment:
I don't think we can state this with authority without at least checking `err` for something that indicates "permission denied".  (Eg, consider how confusing this error message would be if `err` was "connection refused")

I think we just have to add some weasel-words to suggest it's just a guess:
`"failed to set IAM policy on project %q (possibly caused by not having privileges to grant roles/dns.admin role to service account %s): %v"`